### PR TITLE
Improve contributing.md

### PR DESCRIPTION
**Pull request by olsen232**
- More detail on downloading CI artifacts, including scrolling issue
- More detail on DPython_ROOT
- More consistency between windows and posix
- Added some section links at the top

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,11 +36,21 @@ $ git clone https://github.com/koordinates/kart.git
 $ cd kart
 $ git submodule update --init --recursive
 ```
+
+### Setting Python3_ROOT
+
+CMake will try to automatically find a Python3 install on your system with which to build Kart.
+However, to avoid any issues caused by different Python implementations, it is recommended to force CMake
+to use Python 3.10, by supplying the flag -DPython3_ROOT - for example:
+`-DPython3_ROOT=$(realpath $(which python3.10) | sed 's@/bin/.*@/@')`
```
Comment (rcoup) [inline CONTRIBUTING.md:45]: can also do `-DPython3_EXECUTABLE=$(command -v python3.10)` — dunno which is better? Executable might be easier for users.

**Review by rcoup — APPROVED** [review]
LGTM. Minor suggestion but feel free to ignore.

File: CONTRIBUTING.md (hunk 1 above)
Comment (olsen232) [reply CONTRIBUTING.md:45]: Fine by me as long as this is as powerful as setting DPython3_ROOT, ie, cmake correctly finds the root from the executable and doesn't require you to set any other DPython3 variables

Comment (olsen232) [reply CONTRIBUTING.md]: Seems to work, done

**Pull request merged**